src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -1656,6 +1656,124 @@ public function get_test_https_status() {
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return array The test results.
+	 */
+	public function get_test_page_cache() {
+		$description  = '<p>' . __( 'Page caching enhances the speed and performance of your site by saving and serving static pages instead of calling for a page every time a user visits.' ) . '</p>';
+		$description .= '<p>' . __( 'Page caching is detected by looking for an active page caching plugin as well as making three requests to the homepage and looking for one or more of the following HTTP client caching response headers:' ) . '</p>';
+		$description .= '<code>' . implode( '</code>, <code>', array_keys( $this->get_page_cache_headers() ) ) . '.</code>';
+
+		$result = array(
+			'badge'       => array(
+				'label' => __( 'Performance' ),
+				'color' => 'green',
+			),
+			'description' => wp_kses_post( $description ),
+			'test'        => 'page_cache',
+			'status'      => 'good',
+			'label'       => '',
+			'actions'     => sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				__( 'https://wordpress.org/support/article/optimization/#Caching' ),
+				__( 'Learn more about page caching' ),
+				/* translators: Accessibility text. */
+				__( '(opens in a new tab)' )
+			),
+		);
+
+		$page_cache_detail = $this->get_page_cache_detail();
+
+		if ( is_wp_error( $page_cache_detail ) ) {
+			$result['badge']['color'] = 'orange';
+			$result['label']          = __( 'Unable to detect the presence of page caching' );
+			$result['status']         = 'recommended';
+			$error_info               = sprintf(
+			/* translators: 1 is error message, 2 is error code */
+				__( 'Unable to detect page caching due to possible loopback request problem. Please verify that the loopback request test is passing. Error: %1$s (Code: %2$s)' ),
+				$page_cache_detail->get_error_message(),
+				$page_cache_detail->get_error_code()
+			);
+			$result['description'] = wp_kses_post( "<p>$error_info</p>" ) . $result['description'];
+			return $result;
+		}
+
+		$result['status'] = $page_cache_detail['status'];
+
+		switch ( $page_cache_detail['status'] ) {
+			case 'recommended':
+				$result['badge']['color'] = 'orange';
+				$result['label']          = __( 'Page caching is not detected but the server response time is OK' );
+				break;
+			case 'good':
+				$result['badge']['color'] = 'green';
+				$result['label']          = __( 'Page caching is detected and the server response time is good' );
+				break;
+			default:
+				$result['badge']['color'] = 'red';
+				if ( empty( $page_cache_detail['headers'] ) && ! $page_cache_detail['advanced_cache_present'] ) {
+					$result['label'] = __( 'Page caching is not detected and the server response time is slow' );
+				} else {
+					$result['label'] = __( 'Page caching is detected but the server response time is still slow' );
+				}
```

Comment:
I'm not a fan of changing the badge colors, the color currently reflect the area the test belongs to, and not the status of the checks.

Instead, if something is considered so bad we would give it a red angry label, it should have a status of `critical` (this is also important because of accessibility, we can't lean only on colors to indicate status, so you would need different labels as well if you were to use colors here).

Reply:
Hello, @Clorith I am re-checking the code. 
So your suggestion is to set the color to the area it belongs to blue in this case, but then, how do we enforce action if not asking for attention? There are certain levels of warning here, and to me, it makes sense to use different colors for different actions. 

As in the other SH ticket, this is being ported from the performance plugin and we agreed on using such a schematic. I'm tagging @felixarntz @spacedmonkey from the performance group so we can decide all together.

Reply:
Well, using just a color will not pass accessibility review (this was something we had to change with the Health Check plugin when it was merged to core as well, as I had done the exact same thing, this is to be mindful of those with colorblindness).

The status should indicate an action is recommended/needed though, if not you may end up with something "hidden" in the passed checks section, because there was a desire to use badge colors over test statuses :)

Reply:
@Clorith thanks for the clarification. So just to be on the same page. 
We must use the same badge color (blue) for performance, whatever the status is, except for the criticals that we may use a red one. 
If I understood correctly, I will change this and the other PR.

Reply:
@Clorith fixed.